packages/amplify_api/android/src/test/kotlin/com/amazonaws/amplify/amplify_api/GraphQLApiUnitTests.kt

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.amplify.amplify_api
+
+import com.amazonaws.amplify.amplify_api.types.FlutterApiErrorMessage
+import com.amplifyframework.api.ApiCategory
+import com.amplifyframework.api.ApiException
+import com.amplifyframework.api.aws.GsonVariablesSerializer
+import com.amplifyframework.api.graphql.GraphQLOperation
+import com.amplifyframework.api.graphql.GraphQLRequest
+import com.amplifyframework.api.graphql.GraphQLResponse
+import com.amplifyframework.api.graphql.SimpleGraphQLRequest
+import com.amplifyframework.core.Amplify
+import com.amplifyframework.core.Consumer
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.*
+import org.robolectric.RobolectricTestRunner
+import java.lang.reflect.Field
+import java.lang.reflect.Modifier
+
+@RunWith(RobolectricTestRunner::class)
+class GraphQLApiUnitTests {
+    lateinit var flutterPlugin: AmplifyApiPlugin
+    private var mockApi = mock(ApiCategory::class.java)
+    private val mockResult: MethodChannel.Result = mock(MethodChannel.Result::class.java)
+    private val mockGraphQLOperation = mock(GraphQLOperation::class.java)
+
+    @Before
+    fun setup() {
+        flutterPlugin = AmplifyApiPlugin()
+        setFinalStatic(Amplify::class.java.getDeclaredField("API"), mockApi)
+    }
+
+    @Test
+    fun test_query_returns_success() {
+        val testRequest = HashMap<String, Any>();
+        testRequest["document"] = """query MyQuery {
+            listBlogs {
+                items {
+                id
+                name
+                createdAt
+                }
+            }
+        }"""
+        testRequest["variables"] = HashMap<String, Any>();
+
+        val graphQLResponse: GraphQLResponse<String> = GraphQLResponse("result", null)
+
+        doAnswer { invocation ->
+            Assert.assertEquals(
```

Comment:
Instead of asserting in your invocation lambda, you should:
1. Tighten the `when` clause, so that the answer only fires when this condition is met
2. Use a `verify` call towards the end of the test method, that captures the input value using an `ArgumentCaptor`.